MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -131,6 +142,38 @@ open class RouteProgress: NSObject {
         self.legIndex = legIndex
         super.init()
         currentLegProgress = RouteLegProgress(leg: currentLeg, stepIndex: 0, alertLevel: alertLevel)
+        
+        var maneuverCoordinateIndex = 0
+        for (legIndex, leg) in route.legs.enumerated() {
+            
+            congestionTimesPerStep.append([])
+            
+            /// An index into the route’s coordinates and congestionTravelTimesSegmentsByStep that corresponds to a step’s maneuver location.
+            var congestionTravelTimesSegmentsByLeg: [[TimedCongestionLevel]] = []
+            
+            if let segmentCongestionLevels = leg.segmentCongestionLevels, let expectedSegmentTravelTimes = leg.expectedSegmentTravelTimes  {
+                for (stepIndex, step) in leg.steps.enumerated() {
+                    let stepCoordinatesCount = Int(step.coordinateCount) + maneuverCoordinateIndex - 1
+                    
+                    let congestionSegment = Array(segmentCongestionLevels[maneuverCoordinateIndex-stepIndex..<stepCoordinatesCount - stepIndex])
```

Comment:
Can you answer [this question](https://github.com/mapbox/mapbox-navigation-ios/pull/403#discussion_r129901384)?